comp_loinc/build.py

```diff
@@ -1,19 +1,66 @@
-import typer
-from typing import List
-from ingest.part_ingest import PartOntology
-from ingest.code_ingest import CodeIngest
+"""CLI
+
+TODO's
+  1. Flesh out documentation for each param for each command.
+  2. Change args from positional to named, e.g.: https://typer.tiangolo.com/tutorial/parameter-types/number/
+    i. if i can't make them required, then i can raise an err if any are missing
+"""
 import os
 import subprocess
+from pathlib import Path
+
+import typer
+
+try:
+    from comp_loinc.ingest.part_ingest import PartOntology
+    from comp_loinc.ingest.code_ingest import CodeIngest
+except ModuleNotFoundError:
+    from ingest.part_ingest import PartOntology
+    from ingest.code_ingest import CodeIngest
 
-app = typer.Typer()
 
+app = typer.Typer(help='CompLOINC. A tool for creating an OWL version of LOINC.')
+# TODO: PR commment: Does he want this moved to a config? Or no defaults at all?
+PROJECT_DIR = os.path.join(os.path.dirname(__file__), '..')  # todo double check
+DEFAULTS = {
+    'schema_file': os.path.join(PROJECT_DIR, 'model/schema/part_schema.yaml'),
+    'part_primary_file': os.path.join(PROJECT_DIR, 'local_data/LoincPartLink_Primary.csv'),
+    'part_supplementary_file': os.path.join(PROJECT_DIR, 'local_data/LoincPartLink_Supplementary.csv'),
+    'hierarchy_file': os.path.join(PROJECT_DIR, 'local_data/CHEM_HIERARCHY_LPL_DATA.xlsx'),
+    'code_file': os.path.join(PROJECT_DIR, 'model/schema/code_schema.yaml'),
+    'composed_classes_data_file': os.path.join(PROJECT_DIR, 'data/composed_classes_data.yaml'),
+    'owl_directory': os.path.join(PROJECT_DIR, 'data/output/'),
+    'merged_owl': os.path.join(PROJECT_DIR, 'merged_loinc.owl'),
+    'owl_reasoner': os.path.join(PROJECT_DIR, 'elk'),
+    'output.parts': os.path.join(PROJECT_DIR, 'data/output/part_classes.owl'),
+    'output.codes': os.path.join(PROJECT_DIR, 'data/output/code_classes_cli.owl'),
+    'output.composed': os.path.join(PROJECT_DIR, 'data/output/composed_component_classes.owl'),
+    'output.merge': os.path.join(PROJECT_DIR, 'data/output/merged_loinc.owl'),
+    'output.reason': os.path.join(PROJECT_DIR, 'merged_reasponed_loinc.owl'),
+}
+# TODO: PR comment: https://click.palletsprojects.com/en/8.0.x/parameters/
+# path_type
+# path_type (Optional[Type]) – Convert the incoming path value to this type. If None, keep Python’s default, which is str. Useful to convert to pathlib.Path.
+# readable
+# readable (bool) – The file or directory must be readable.
+# writable
+# writable (bool) – The file or directory must be writable.
+
+# TODO: PR comment: does he want Option instead of Argument? I think that's how you do named
 @app.command(name='parts')
 def build_part_ontology(
-        schema_file: str,
-        part_primary_file: str,
-        part_supplementary_file: str,
-        hierarchy_file: str,
-        output: str):
+        schema_file: Path = typer.Argument(default=DEFAULTS['schema_file'], resolve_path=True, exists=True),
```

Comment:
Updated typing and [parameterization for args](https://click.palletsprojects.com/en/8.0.x/parameters/).

Additional things we could add to `typer.Argument()` if we like:
> `path_type (Optional[Type])` – Convert the incoming path value to this type. If `None`, keep Python’s default, which is `str`. Useful to convert to `pathlib.Path`.

> `readable (bool)` – The file or directory must be readable.